src/lib.rs

```diff
@@ -3939,10 +3998,14 @@ pub struct SemanticTokens {
 
     /**
      * The actual tokens. For a detailed description about how the data is
-     * structured pls see
+     * structured please see
      * https://github.com/microsoft/vscode-extension-samples/blob/5ae1f7787122812dcc84e37427ca90af5ee09f14/semantic-tokens-sample/vscode.proposed.d.ts#L71
      */
-    pub data: Vec<u32>,
+    #[serde(
+        deserialize_with = "SemanticToken::deserialize_tokens",
```

Comment:
Nit: If you drop the `tokens` suffixes you could use https://serde.rs/field-attrs.html#with `#[serde(with = "SemanticToken")]` (though naming might be clearer if they are in a submodule like `semantic_tokens` `#[serde(with = "semantic_tokens")]`

Reply:
I had to fight with serde to get the `Option<Vec<SemanticToken>>` to ser/de properly so I think it's clearer if I leave these in.